core/src/main/java/org/infinispan/globalstate/impl/GlobalConfigurationManagerImpl.java

```diff
@@ -85,18 +91,42 @@ void start() {
       parserRegistry = new ParserRegistry();
 
       localConfigurationManager.initialize(cacheManager, configurationManager, executorService);
+
+      // Load any state we previously had
```

Comment:
This is kind of confusing, both the local persistent state and the state cache can be said to hold existing configuration state. Could you rename the variables a bit to make it clear where each came from?

Reply:
I have renamed things to make things clearer